### PR TITLE
Use AppVeyor

### PR DESCRIPTION
**Pull request by nox**
[Review on Reviewable](https://reviewable.io/reviews/servo/heapsize/45)

### CODE REVIEW

**Comment by nox** [comment]
This is blocked by #44 because heapsize is broken with last nightlies.

**Comment by larsbergstrom** [comment]
r=me once #44 lands

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  matrix:
+  - TARGET: x86_64-pc-windows-msvc
+  - TARGET: i686-pc-windows-msvc
+  - TARGET: i686-pc-windows-gnu
+
+install:
+  - ps: Start-FileDownload "https://static.rust-lang.org/dist/rust-nightly-${env:TARGET}.exe"
+  - rust-nightly-%TARGET%.exe /VERYSILENT /NORESTART /DIR="C:\Program Files (x86)\Rust"
+  - SET PATH=%PATH%;C:\Program Files (x86)\Rust\bin
+  - SET PATH=%PATH%;C:\MinGW\bin
```
Comment (retep998) [inline appveyor.yml]: This line is not needed unless you intend to actually use MinGW. `-gnu` Rust comes with a bundled MinGW which should be sufficient for most needs.

File: src/lib.rs [hunk 2]
```diff
@@ -18,38 +23,54 @@ use std::sync::Arc;
...
+                  target_arch = "mipsel",
+                  target_arch = "powerpc")))]
+    const MIN_ALIGN: usize = 8;
+    #[cfg(all(any(target_arch = "x86_64",
+                  target_arch = "aarch64")))]
+    const MIN_ALIGN: usize = 16;
+
+    let heap = GetProcessHeap();
+
+    if ptr as usize % (MIN_ALIGN * 2) != 0 && HeapValidate(heap, 0, ptr) == 0 {
```
Comment (retep998) [inline src/lib.rs]: This should actually be `ptr as usize % (MIN_ALIGN * 2) == 0`. Personally I'd just depend entirely on `HeapValidate`.

File: src/lib.rs [hunk 3]
```diff
@@ -18,38 +23,53 @@ use std::sync::Arc;
...
+    // Copied from Rust's liballoc_system.
+    #[cfg(all(any(target_arch = "x86",
+                  target_arch = "arm",
+                  target_arch = "mips",
+                  target_arch = "mipsel",
+                  target_arch = "powerpc")))]
+    const MIN_ALIGN: usize = 8;
+    #[cfg(all(any(target_arch = "x86_64",
+                  target_arch = "aarch64")))]
+    const MIN_ALIGN: usize = 16;
```
Comment (retep998) [inline src/lib.rs]: You don't actually use `MIN_ALIGN` anywhere in this function.

**Comment by nox** [comment]
@bors-servo r-

AppVeyor unhappy.

**Comment by nox** [comment]
Ah no, never mind.

**Pull request merged**